src/lib/hooks/transactions/retry.ts

```diff
@@ -6,21 +6,15 @@ function waitRandom(min: number, max: number): Promise<void> {
   return wait(min + Math.round(Math.random() * Math.max(0, max - min)))
 }
 
-/**
- * This error is thrown if the function is cancelled before completing
- */
-class CancelledError extends Error {
-  public isCancelledError = true as const
-  constructor() {
-    super('Cancelled')
-  }
+/** Thrown if the function is canceled before resolving. */
+export class CanceledError extends Error {
+  name = 'CanceledError'
+  message = 'Retryable was canceled'
```

Comment:
this diff seems fine but would have preferred it go in standalone

Reply:
seems unrelated afaict

Reply:
it's related in that this PR touches the consumer of the retry logic, but that's a stretch. but it feels harder to extricate, because the consumer is coupled to the implementation, so the new PR would necessarily have merge conflicts once this one went in.